packages/core/src/editing/codegen/preview.ts

```diff
@@ -0,0 +1,319 @@
+  name: typeof DESIGN_LIBRARY_COMPONENT_PROPS_EVENT_NAME;
+  message: {
+    uid: string;
+    fields: ComponentFields;
+    parameters: ComponentParams;
+  };
+}
+
+/**
+ * Represents an event indicating the preview error to be sent to design library.
+ */
+export interface DesignLibraryComponentPreviewErrorEvent {
+  name: typeof DESIGN_LIBRARY_COMPONENT_PREVIEW_ERROR_EVENT_NAME;
+  message: {
+    uid: string;
+    error: unknown;
+    type: DesignLibraryPreviewError;
+  };
+}
+
+/**
+ * Enumeration of error types for the design library preview.
+ */
+export enum DesignLibraryPreviewError {
+  /**
+   * Error occurred during component rendering.
+   */
+  Render = 'render',
+  /**
+   * Error occurred during component and event handlers initialization.
+   */
+  RenderInit = 'render-init',
+}
+
+/**
+ * Builds the component dependencies from the component imports and the import map.
+ * @param {ComponentImport[]} componentImports - The component imports.
+ * @param {ImportEntry[]} importMap - The import map to be used for the component.
+ * @returns {Array<{ name: string; value: unknown }>} The component dependencies.
+ */
+export function buildComponentDependencies(
+  componentImports: ComponentImport[],
+  importMap: ImportEntry[]
+): Array<{ name: string; value: unknown }> {
+  return componentImports.flatMap((componentImport) => {
+    const moduleEntry = importMap.find((entry) => entry.module === componentImport.module);
+
+    const exportEntry = moduleEntry?.exports.find((exp) => exp.name === componentImport.export);
+
+    return exportEntry
+      ? {
+          name: componentImport.alias,
+          value: exportEntry.value,
+        }
+      : [];
+  });
+}
+
+/**
+ * Adds the browser-side event handler for 'component:generation:component-preview' message used in Design Library
+ * The event should contain the component code, styles and imports.
+ * @param {ImportEntry[]} importMap - The import map to be used for the component.
+ * @param {Function} callback callback to be called after component is received
+ */
+export const addComponentPreviewHandler = (
+  importMap: ImportEntry[],
+  callback: (error: unknown | null, Component: unknown) => void
+) => {
+  if (!window) return;
+
+  const handler = (e: MessageEvent) => {
+    const eventArgs: ComponentPreviewEventArgs = e.data;
+
+    try {
+      if (!e.origin || !eventArgs || eventArgs.name !== 'component:generation:component-preview') {
+        // avoid extra noise in logs
+        if (!validateOrigin(e)) {
+          console.debug(
+            'Component Library: event skipped - invalid origin: message %s from origin %s',
+            eventArgs.name,
+            e.origin
+          );
+        }
+
+        return;
+      }
+
+      console.debug('Component Library: message received', eventArgs);
+
+      const { message } = eventArgs;
+
+      const componentDepencencies = buildComponentDependencies(message.imports, importMap);
```

Comment:
What if incoming message contains an import dependency that is not present in the import map? I suppose currently code will fail with prop of undefined exception or similarly. 
What if we consider it as error or warning here, to provide better context to user and DL app?